### ISSUE TITLE
`machine ip` panics for no active host set

### ISSUE THREAD

**Issue by ahmetalpbalkan** [docker/machine#222]
I'm running on OS X, built from master 0b98347.

`machine ip` fails with "invalid memory address or nil pointer deference" panic.

Same goes for `machine [stop|start|ssh|...]` in the state I am in.

``` sh
$ machine ls
NAME      ACTIVE   DRIVER   STATE     URL
myazure            azure    Running   tcp://docker-machine.cloudapp.net:4243

$ machine ip
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x771f]

goroutine 16 [running]:
runtime.panic(0x592fc0, 0x908844)
	/usr/src/go/src/pkg/runtime/panic.c:279 +0xf5
main.func·004(0x208a8c240)
	/go/src/github.com/docker/machine/commands.go:139 +0x1bf
github.com/codegangsta/cli.Command.Run(0x616cb0, 0x2, 0x0, 0x0, 0x674790, 0x1f, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/docker/machine/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:101 +0xc6f
github.com/codegangsta/cli.(*App).Run(0x208a8c0c0, 0x2089d8000, 0x2, 0x2, 0x0, 0x0)
	/go/src/github.com/docker/machine/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:130 +0x953
main.main()
	/go/src/github.com/docker/machine/main.go:36 +0x310

goroutine 19 [finalizer wait]:
runtime.park(0x1de30, 0x90c658, 0x90a909)
	/usr/src/go/src/pkg/runtime/proc.c:1369 +0x89
runtime.parkunlock(0x90c658, 0x90a909)
	/usr/src/go/src/pkg/runtime/proc.c:1385 +0x3b
runfinq()
	/usr/src/go/src/pkg/runtime/mgc0.c:2644 +0xcf
runtime.goexit()
	/usr/src/go/src/pkg/runtime/proc.c:1445

goroutine 20 [syscall]:
os/signal.loop()
	/usr/src/go/src/pkg/os/signal/signal_unix.go:21 +0x1e
created by os/signal.init·1
	/usr/src/go/src/pkg/os/signal/signal_unix.go:27 +0x32

goroutine 21 [runnable]:
text/template/parse.lexText(0x208a16400, 0x7278d8)
	/usr/src/go/src/pkg/text/template/parse/lex.go:228 +0x381
text/template/parse.(*lexer).run(0x208a16400)
	/usr/src/go/src/pkg/text/template/parse/lex.go:198 +0x40
created by text/template/parse.lex
	/usr/src/go/src/pkg/text/template/parse/lex.go:191 +0x112
```

**Comment by ehazlett** [comment]
Thanks.  I'm also getting some errors.  Investigating.

**Comment by nathanleclaire** [comment]
Ah!  I just ran into this as well (with virtualbox hosts!).  I'm not sure how, but the active host is getting unset.  Notice that in your output the `*` is missing.

**Comment by nathanleclaire** [comment]
Update: Not sure if that is the issue here but if you `machine rm` the active host there will not be another active host chosen and machine will freak out on some commands like `machine ip` which assume that they will be run against an active host.

**Comment by ahmetalpbalkan** [comment]
@nathanleclaire okay that makes sense, my `machine create` command failed in the middle.

**Comment by ehazlett** [comment]
Perhaps we should report an error if a command is ran and there is no active machine?

**Comment by nathanleclaire** [comment]
> Perhaps we should report an error if a command is ran and there is no active machine?

Yes, and should probably : 

- choose a fallback if you `rm` the active one.
- not switch or lose active hosts when a `create` fails


Note that `machine ip` if you have no hosts at all will still panic.

```console
$ machine ls
NAME   ACTIVE   DRIVER   STATE   URL

$ machine ip
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x7c4f]

goroutine 16 [running]:
runtime.panic(0x592f20, 0x908844)
        /usr/src/go/src/pkg/runtime/panic.c:279 +0xf5
main.func·004(0x208a8c240)
        /go/src/github.com/docker/machine/commands.go:168 +0x1bf
github.com/codegangsta/cli.Command.Run(0x616af0, 0x2, 0x0, 0x0, 0x6745b0, 0x1f, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/docker/machine/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:101 +0xc6f
github.com/codegangsta/cli.(*App).Run(0x208a8c0c0, 0x2089d8000, 0x2, 0x2, 0x0, 0x0)
        /go/src/github.com/docker/machine/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:130 +0x953
main.main()
        /go/src/github.com/docker/machine/main.go:36 +0x310

goroutine 19 [finalizer wait]:
runtime.park(0x1de10, 0x90c658, 0x90a909)
        /usr/src/go/src/pkg/runtime/proc.c:1369 +0x89
runtime.parkunlock(0x90c658, 0x90a909)
        /usr/src/go/src/pkg/runtime/proc.c:1385 +0x3b
runfinq()
        /usr/src/go/src/pkg/runtime/mgc0.c:2644 +0xcf
runtime.goexit()
        /usr/src/go/src/pkg/runtime/proc.c:1445

goroutine 20 [syscall]:
os/signal.loop()
        /usr/src/go/src/pkg/os/signal/signal_unix.go:21 +0x1e
created by os/signal.init·1
        /usr/src/go/src/pkg/os/signal/signal_unix.go:27 +0x32

goroutine 21 [runnable]:
text/template/parse.lexText(0x208a18400, 0x7276e0)
        /usr/src/go/src/pkg/text/template/parse/lex.go:228 +0x381
text/template/parse.(*lexer).run(0x208a18400)
        /usr/src/go/src/pkg/text/template/parse/lex.go:198 +0x40
created by text/template/parse.lex
        /usr/src/go/src/pkg/text/template/parse/lex.go:191 +0x112
```

**Comment by bfirsh** [comment]
Was this fixed in #262?

**Comment by ehazlett** [comment]
yes.  thanks!  closing.
